src/app/components/task-list/task-list.component.ts

```diff
@@ -17,16 +17,15 @@ import { RemoveButtonComponent } from 'src/app/remove-button/remove-button.compo
 export class TaskListComponent implements OnInit, OnDestroy {
   completedTasksSubscription = Subscription.EMPTY;
   tasksCompleted: number = 0;
+  deleteSubscription = Subscription.EMPTY;
   tasksSubscription = Subscription.EMPTY;
   tasks: Task[] = [];
 
-  constructor(private http: HttpClient, private taskService: TaskService) {}
+  constructor(private http: HttpClient, private taskService: TaskService) { }
 
   ngOnInit(): void {
-    this.taskService.getTasks().pipe(take(1)).subscribe((tasks) => {
-      this.taskService.setTasks(tasks);
-    })
-    this.tasksSubscription = this.taskService.tasksSubject.subscribe((tasks) => {
+    this.taskService.refresh();
+    this.tasksSubscription = this.taskService.tasksSubject.subscribe(tasks => {
```

Comment:
here instead of subscribing constantly you can use an observable (tasks$ in taskService) which will hold the value of the behaviourSubject (you can look up the method asObservable) that way you can use this tasks$ observable in the template where it automatically subscribes/unsubscribes

Reply:
or you can just use the behaviourSubject directly with tasksSubject.asObservable() wherever you need